### PR TITLE
Added RELEASING docs

### PR DESCRIPTION
**Pull request by lemnik**
## Goal
Introduce a simple process for releasing `bugsnag_flutter` to pub.dev

### CODE REVIEW

File: docs/RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Releasing
+=========
+
+- Using GitHub create a new releasing branch from `next`: `releases/v<version number>`
+- Checkout the release branch
+  - Bump the version number: `make VERSION=<version> bump`
+  - Inspect the updated CHANGELOG, README, and version files to ensure they are correct
+  - Open a PR from the release branch to `main`
+- Once merged:
+  - Pull the latest changes into a *fresh* local repository (clone `main`)
```
Comment (tomlongridge) [inline docs/RELEASING.md:10]: Is this always required, or can we rely on `make clean`?

Comment (lemnik) [reply docs/RELEASING.md:10]: We can't completely rely on `make clean` as far as I've seen.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Changelog
 
-## 2.0.0
+## 2.0.0-rc5
```
[the remark this answers is not in the document]
Comment (lemnik) [reply CHANGELOG.md:3]: `pub.dev` will reject the upload if the version number in CHANGELOG doesn't match the one in `pubspec`

[the remark this answers is not in the document]
Comment (tomlongridge) [reply CHANGELOG.md:3]: In that case, shall we add this line to `make bump` (like [bugsnag-cocoa](https://github.com/bugsnag/bugsnag-cocoa/blob/master/Makefile#L176)?) It won't guarantee that we'll include the version as it relies on someone having added the "TBD", but at least it's a hint that it needs doing.

File: docs/RELEASING.md (hunk 1 above)
Comment (tomlongridge) [reply docs/RELEASING.md:10]: I thought that was the idea of #102? What about running `git clean`?

Comment (lemnik) [reply docs/RELEASING.md:10]: That is the idea of #102 but it is very easy for new files / dirs to slip in and accidentally be included in the release. I think a `git clean` would also be an acceptable solution. I'll amend the doc

**Review by tomlongridge — APPROVED** [review]

**Pull request merged**